File: 0078-subsets/0078-subsets.java
class Solution {
    public List<Integer> l;
    public List<List<Integer>> ll;
    
    public List<List<Integer>> subsets(int[] nums) {
        l = new ArrayList<>();
        ll = new ArrayList<>();
        allSets(nums, 0);
        return ll;
    }
    
    public void allSets(int[] nums, int i)
    {
        if (i >= nums.length)
        {
            System.out.println(l.size());
            ll.add(new ArrayList<>(l));
            return;
        }
        
        
        l.add(nums[i]);
        allSets(nums, i + 1);
        
        l.remove(l.size() - 1);
        allSets(nums, i + 1);
    }
}

File: 0078-subsets/0078-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []
        temp = []
        def backtrack(i):
            nonlocal res
            nonlocal temp
            if i >= len(nums):
                res.append(temp.copy())
                return
            temp.append(nums[i])
            backtrack(i + 1)
            temp.pop()
            backtrack(i + 1)
            

        backtrack(0)
        return res
        